invoices/hodl_manager.go

```diff
@@ -0,0 +1,191 @@
+package invoices
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/lightningnetwork/lnd/lntypes"
+)
+
+// HodlManager manages hold invoices.
+type HodlManager struct {
+
+	// Registry is a sub-system which responsible for managing the invoices
+	// in thread-safe manner.
+	registry InvoiceDatabase
```

Comment:
Circling back from the existing `htlcswitch` comment, I don't think it needs all these methods yet. Also, by using a slimmer interface, we can have the `SetPreimage` method (or w/e it's called) also insert into the pre-image cache.

Reply:
Ahh actually nvm, on that second portion. Doing so would start to intermingle locally created/sourced pre-images from those obtained via forwarding. I think this answers my questions in the contract court.

Reply:
First part not applicable anymore, because no more hodl manager.